src/core/shape_inference/include/stft_shape_inference.hpp

```diff
@@ -46,20 +48,23 @@ std::vector<TRShape> shape_infer(const STFT* op,
     if (signal_shape.rank().is_dynamic()) {
         return {signal_shape};
     } else if (!frame_size || !frame_step) {
-        return {TRShape{signal_shape[0], -1, -1, 2}};
+        return {TRShape{signal_shape[0], TDim(ov::util::dim::inf_bound), TDim(ov::util::dim::inf_bound), 2}};
     }
 
     const auto& frame_size_val = (*frame_size)[0];
     const auto& frame_step_val = (*frame_step)[0];
 
-    NODE_SHAPE_INFER_CHECK(op,
-                           input_shapes,
-                           0 < frame_size_val && frame_size_val < signal_shape[1].get_interval().get_max_val(),
-                           "Provided frame size is ",
-                           frame_size_val,
-                           " but must be in range [1, ",
-                           signal_shape[1],
-                           "].");
+    NODE_SHAPE_INFER_CHECK(
+        op,
+        input_shapes,
+        0 < frame_size_val &&
+            (signal_shape[1].is_static() ? static_cast<TDimVal>(frame_size_val) < signal_shape[1].get_length()
+                                         : frame_size_val < signal_shape[1].get_interval().get_max_val()),
```

Comment:
can this check be set outside to some meaningful name. The whole condition will be stringified and hard to read. Add `bool is_frame_size_in_range` ?

Reply:
Sure, done, refactored for the window_shape check as well